app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterView.kt

```diff
@@ -7,47 +7,44 @@ import com.puzzlebench.clean_marvel_kotlin.domain.model.Character
 import com.puzzlebench.clean_marvel_kotlin.presentation.MainActivity
 import com.puzzlebench.clean_marvel_kotlin.presentation.adapter.CharacterAdapter
 import com.puzzlebench.clean_marvel_kotlin.presentation.extension.showToast
+import com.puzzlebench.clean_marvel_kotlin.presentation.fragment.DetailCharacterFragmentDialog
 import kotlinx.android.synthetic.main.activity_main.*
 import java.lang.ref.WeakReference
 
-class CharecterView(activity: MainActivity) {
-    private val activityRef = WeakReference(activity)
+class CharacterView(activity: MainActivity) {
+    private  val activity = activity
```

Comment:
You don't need this reference to the activity, you have the injected reference. Please remove this variable and refactor the constructor as follows:
```
class CharacterView(private val activity: MainActivity) {
```